### ISSUE TITLE
add now_or_never

### ISSUE THREAD

**Issue by dvc94ch** [smol-rs/futures-lite#32]
this has stopped me from using futures-lite a couple of times now. incredibly useful function.

**Comment by Koxiaet** [comment]
It currently exists as `futures_lite::future::poll_once`, although it's not a method like `.now_or_never()`.

**Comment by dvc94ch** [comment]
No, that's not the same. It requires you to construct a waker, and if you look at the futures NoopWaker it's about 200loc.

**Comment by Koxiaet** [comment]
Oh, sorry, I didn't realize. But it can still be useful when you are already in an async context, which is often the case.
